srk/src/chc.ml

```diff
@@ -244,120 +492,119 @@ module ChcSrkZ3 = struct
     | REAL_SORT -> `TyReal
     | INT_SORT -> `TyInt
     | BOOL_SORT -> `TyBool
-    | _ -> failwith "TODO: allow function types"
+    (*| ARRAY_SORT -> `TyArr*)
+    |_ -> failwith "function types not currently supported in rel param"
 
-  (* Creates a relation atom from a z3 predicate in which each argument
-   * to the predicate is an integer. Replaces integer [i] with value
-   * located at key [i] in table [ind_to_sym] when such a key exists *)
-  let rel_atom_of_z3 srk fp ind_to_sym rsym_to_int names z3pred =
+  let rel_atom_of_z3 srk fp ind_to_sym rsym_to_int z3pred =
```

Comment:
Why can't you use SrkZ3.formula_of_z3?

Reply:
I believe that should work. I would still need some kind of rel_atom_of_formula_predicate function though. This is because my relation atoms are much more restricted than those allowed in smt2, even than those produced by seahorn. I require that arguments to a relation symbol be variable symbols.